src/Order/Bulk.php

```diff
@@ -162,7 +162,12 @@ public function bulk_action_change_shipping_options( $redirect, $doaction, $obje
 					);
 				}
 
-				if ( ! $have_label_file && $match_shipping_zones ) {
+				if ( ! $have_label_file && 'PICKUP' === $zone ) {
+					$meta = $order->get_meta( $this->meta_name );
+					$meta['backend'] = $selected_shipping_options ;
+					$order->update_meta_data( $this->meta_name, $meta );
+					$order->save();
+				}elseif ( ! $have_label_file && $match_shipping_zones ) {
```

Comment:
I don't understand the purpose of this change

`! $have_label_file && 'PICKUP' === $zone` and `! $have_label_file && $match_shipping_zones` will be always the same, due to this change :
`$match_shipping_zones = $zone === $this->get_shipping_zone( $order ) || 'PICKUP' === $zone;`

Reply:
@Abdalsalaam 
```
if ( ! $have_label_file && 'PICKUP' === $zone ) {
					$meta = $order->get_meta( $this->meta_name );
					$meta['backend'] = $selected_shipping_options ;
					$order->update_meta_data( $this->meta_name, $meta );
					$order->save();
				}
```
				this will be applied if the order has drop-off option from frontend to make sure it will not remove the frontend data